### ISSUE TITLE
Guided, Supervised Topic Modeling takes too long 

### ISSUE THREAD

**Issue by TAsUjxnMIL** [MaartenGr/BERTopic#458]
Hello Maarten,
I try to execute the Guided Topic Modeling. This does take too much time, after an hour the process is still not finished. Is there any possibility too speed it up? Is this even normal that it takes too much time? 
Edit: Reducing reducing the number of documents, I get the error: 
**No loop matching the specified signature and casting was found for ufunc add**

I also tried to do the Supervised Topic Modeling.
`topic_model = BERTopic(verbose=True).fit(docs, y=categories)`
Do I have to execute the fit_transform after executing this line above with the same documents?
I have a corpus and each document is labeled with its topics. How to set the y variable with the labeling?

**Comment by MaartenGr** [comment]
> No loop matching the specified signature and casting was found for ufunc add

Fortunately, others have experienced this issue and there is a known fix for it [here](https://github.com/MaartenGr/BERTopic/issues/202#issuecomment-904494090). It is an issue of having an incompatible numpy/numba combination that should be resolved by changing their respective versions. 

> Do I have to execute the fit_transform after executing this line above with the same documents?

I would advise adhering to the following structure when training BERTopic:

```python
topic_model = BERTopic(verbose=True)
topics, probs = topic_model.fit_transform(docs, y=categories)
```

That way, you can train the model and get the output without running `fit_transform` after `fit`. 

> I have a corpus and each document is labeled with its topics. How to set the y variable with the labeling?

There are two ways in approaching this. First, you can use the topic descriptions as input using guided topic modeling as described in the tutorial [here](https://maartengr.github.io/BERTopic/getting_started/guided/guided.html). Second, you can use those labels as the `y` variable in semi-supervised topic modeling [here](https://maartengr.github.io/BERTopic/getting_started/supervised/supervised.html).

**Comment by TAsUjxnMIL** [comment]
Okay I understand, but in semi-supervised topic modeling do the labels need to be in the same order like the documents, to assign them to each other or is the order not of relevance?

**Comment by MaartenGr** [comment]
Yes, they need to be in the same order as the documents.

**Comment by TAsUjxnMIL** [comment]
Okay thank you very much!
The problem:
**No loop matching the specified signature and casting was found for ufunc ad**d
could also be solved by changing the package versions of numpy and numba.
